addons/product/models/product_pricelist.py

```diff
@@ -111,16 +111,16 @@ def _get_products_price(self, products, quantity, uom=None, date=False):
             ).items()
         }
 
-    def _get_product_price(self, product, quantity, uom=None, date=False):
+    def _get_product_price(self, product, quantity, currency=None, uom=None, date=False):
         """Compute the pricelist price for the specified product, qty & uom.
 
-        Note: self.ensure_one()
+        Note: self and self.ensure_one()
 
         :returns: unit price of the product, considering pricelist rules
         :rtype: float
         """
-        self.ensure_one()
-        return self._compute_price_rule(product, quantity, uom=uom, date=date)[product.id][0]
+        self and self.ensure_one()
+        return self._compute_price_rule(product, quantity, currency=currency, uom=uom, date=date)[product.id][0]
 
     def _get_product_price_rule(self, product, quantity, uom=None, date=False):
         """Compute the pricelist price & rule for the specified product, qty & uom.
```

Comment:
Why is this method the only one where currency is not expected as param? seems strange :thinking:

Reply:
Reading it again, only `_get_product_price` and `_compute_price_rule` needs the currency as a param, because they are called from the POS, where without a pricelist, the currency could be the one from the POS and not from the company. All the other methods are either called for sale or website, where when there isn't a pricelist, the currency is the one from the company env, so I felt no need to pass it as a parameter

Reply:
Because I forget to take out the currency param in this method?  '^^

Only `_get_product_price` and  `_compute_price_rule` should have a currency in the parmaters as they are called in the POS, for all the other methods, we can get it from the company when the recordset pricelist is empty

Reply:
Well, I think I prefer to have a consistent API between the different methods, even if the currency param is not strictly needed in our code for those methods :thinking:.

Reply:
Agreed, I will add it as soon as we get the green light for this task to go further =]